### PR TITLE
add proper permissions to config and template files

### PR DESCRIPTION
**Pull request by aj-jester**
Currently the permissions on the config files where the token is stored is world readable. This PR makes the default `0660` and also refactored how user/group is created inside the module so if the user/group is created outside the module we can still pass the values to the resources inside the module.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -49,14 +52,21 @@
 # [*data_dir*]
 #   Path to a directory to create to hold some data. Defaults to ''
 #
+# [*user*]
+#   Name of a user to use for dir and file perms. Defaults to `false`.
```
Comment (gdhbashton) [inline manifests/init.pp]: Defaults to `consul-template`

File: manifests/init.pp [hunk 2]
```diff
@@ -49,14 +52,21 @@
...
+#
+# [*group*]
+#   Name of a group to use for dir and file perms. Defaults to `false`.
```
Comment (gdhbashton) [inline manifests/init.pp]: Defaults to `consul-template`

**Comment by gdhbashton** [comment]
Yep I like this - just a couple of cosmetic typos and we're good to merge.

**Comment by aj-jester** [comment]
Good catch! fixed.

**Pull request merged**